client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -1176,6 +1176,26 @@ Undocumented.prototype.ebanxConfiguration = function( query, fn ) {
 	return this.wpcom.req.get( '/me/ebanx-configuration', query, fn );
 };
 
+/**
+ * GET emergent paywall iframe client configuration
+ *
+ * @param {string} countryCode - user's country code
+ * @param {object} cart - current cart object. See: client/lib/cart/store/index.js
+ * @param {Function} fn The callback function
+ * @api public
+ *
+ * @returns {Promise} promise
+ */
+Undocumented.prototype.emergentPaywallConfiguration = function( countryCode, cart, fn ) {
+	debug( '/me/emergent-paywall-configuration query' );
+
+	return this.wpcom.req.post(
+		'/me/emergent-paywall-configuration',
+		{ country: countryCode, cart },
+		fn
+	);
+};
```

Comment:
We also want to avoid adding more functionality into `wpcom.js` (sorry - we need more alarms to fire when people add code there)

Instead we have the [data layer](https://wpcalypso.wordpress.com/devdocs/client/state/data-layer/README.md) built up so that we can handle not only feature-specific needs but also app-wide needs for data requests and updates. The data layer is coupled through _data_ as well and not through promises, has request auto-retry behaviors, and is the platform upon which we have been moving for the past year or two and plan on moving towards the future.

You can read about making calls to authenticated WordPress.com API endpoints in the [guide](https://wpcalypso.wordpress.com/devdocs/client/state/data-layer/wpcom-http/README.md)

Would you mind also circling back around here to remove the new `Undocumented` code and move it into the data layer? I'd be happy to review your PRs and help or do screenshares if you would like to see how we can do that. Usually the only code in `wpcom.js` is legacy code awaiting to be removed.

Reply:
Thanks for the comments and taking the time to explain @dmsnell 

Everything makes sense. I'll make a second pass at this component in light of your suggestions, probably next week, and ping you for review.

> we need more alarms to fire when people add code there

What could we do to help contributors here? It's low-tech, but even a comment block at the top of the offending files would help. Even if to save valuable minutes on a retroactive PR review.

Reply:
Thanks again! Probably a comment block would be ineffective. We can write things everywhere but as long as people look at other code as an example to copy then we'll get this stuff popping up even when our guides and docs are clear.

I think we can explore setting up some alerts in Github with code ownership.

Thanks for being so open and willing with this stuff - that's really encouraging! Please don't hesitate to lean on me for anything in this process 👍

Reply:
No worries. Appreciate your offer of assistance, on which I will most happily lean! 

🙇